app/src/main/java/com/codesoom/assignment/service/TaskService.java

```diff
@@ -0,0 +1,56 @@
+package com.codesoom.assignment.service;
+
+import com.codesoom.assignment.exceptions.TaskNotFoundException;
+import com.codesoom.assignment.model.Task;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.*;
+
+import java.sql.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+    private List<Task> tasks = new ArrayList<>();
+    private Long id = 0L;
+
+    public List<Task> getTasks(){
+        return tasks;
+    }
+
+    public Task getTask(Long id){
+        Task task = findTask(id);
+        return task;
+    }
+
+    public synchronized Task createTask(Task task){
+        task.setId(id++);
+        tasks.add(task);
+
+        return task;
+    }
+
+    public Task updateTask(Task task){
+        Task updateTask = findTask(task.getId());
+        updateTask.setTitle(task.getTitle());
+
+        return task;
+    }
+
+    public Task deleteTask(Long id){
+        Task deleteTask = findTask(id);
+        tasks.remove(deleteTask);
+
+        return deleteTask;
+    }
+
+    public Task findTask(Long id){
+        return Optional.ofNullable(
+                tasks.stream()
+                     .filter(t -> t.getId().equals(id))
+                     .findFirst()
+                     .orElse(null))
+                     .orElseThrow(TaskNotFoundException::new);
+    }
```

Comment:
할 일을 찾은 경우에만 값을 반환하고 만약 없는 경우에는 예외를 던지고 싶으셨던거죠.
`findFirst`메서드가 Optional 타입을 반환하기 때문에 감싸주지 않아도 됩니다.

```suggestion
    public Task findTask(Long id){
                tasks.stream()
                     .filter(t -> t.getId().equals(id))
                     .findFirst()
                     .orElseThrow(TaskNotFoundException::new);
    }
```

Reply:
피드백 감사합니다! stream 사용에 대해 좀 더 공부해 봐야겠습니다.
아 그리고 throw 부분 작성 중에 찾아본 결과 () - > TaskNotFoundExcepiton 으로 대부분 작업을 
하던데, tool에서는 TaskNotFoundException::new 로 피드백을 주어 처리 해 보았습니다만,
요 부분에 대해서는 제가 검색을 잘 못하는 것인지, 찾을 수가 없었는데, 
저 부분을 뭐라고 해야 할까요?